db.js

```diff
@@ -0,0 +1,36 @@
+const { MongoClient } = require("mongodb");
+
+module.exports = async function () {
+
+  const url = "mongodb://localhost:27017";
+
+  // The name of the database we will create in Mongo
+  const dbName = "messenger";
+
+  //Connect to the server
+  const client = await MongoClient.connect(url);
+
+  // Open a connection to the database
+  const db = client.db(dbName);
+
+  // initialize our collections
+  db.createCollection("users", {
+    validator: {
+      // Left as an exercise - see note below
+    },
+  });
+
+  db.createCollection("rooms", {
+    validator: {
+      // Left as an exercise - see note below
+    },
+  });
+
+  db.createCollection("messages", {
+    validator: {
+      // Left as an exercise - see note below
+    },
+  });
```

Comment:
Instead, let's initialize the collections this way - that way if the schemas change in code, they will be updated in Mongo too.

```js
// initialize our collections
db.createCollection("users");
db.createCollection("rooms");
db.createCollection("messages");

// set the schema validators
db.command({ collMod: "users", validator: { $jsonSchema: userSchema } });
db.command({ collMod: "rooms", validator: { $jsonSchema: roomSchema } });
db.command({ collMod: "messages", validator: { $jsonSchema: messageSchema } });
```

Reply:
What is collMod and validator?